azurerm/resource_arm_mariadb_database.go

```diff
@@ -0,0 +1,176 @@
+				),
+			},
+
+			"charset": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+				ValidateFunc: validation.StringMatch(
+					regexp.MustCompile("^[a-z0-9]{3,8}$"),
+					"charset must be 3 - 8 characters long, and contain only lowercase letters and numbers.",
+				),
+			},
+
+			"collation": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validateCollation(),
+			},
+		},
+	}
+}
+
+func resourceArmMariaDbDatabaseCreateOrUpdate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient).mariadbDatabasesClient
+	ctx := meta.(*ArmClient).StopContext
+
+	log.Printf("[INFO] preparing arguments for AzureRM MariaDB Database creation.")
+
+	name := d.Get("name").(string)
+	resGroup := d.Get("resource_group_name").(string)
+	serverName := d.Get("server_name").(string)
+
+	charset := d.Get("charset").(string)
+	collation := d.Get("collation").(string)
+
+	properties := mariadb.Database{
+		DatabaseProperties: &mariadb.DatabaseProperties{
+			Charset:   utils.String(charset),
+			Collation: utils.String(collation),
+		},
+	}
+
+	future, err := client.CreateOrUpdate(ctx, resGroup, serverName, name, properties)
+	if err != nil {
+		return fmt.Errorf("Error creating Maria DB %q (Resource Group %q): %+v", name, resGroup, err)
+	}
+
+	if err = future.WaitForCompletionRef(ctx, client.Client); err != nil {
+		return fmt.Errorf("Error waiting for completion of Maria DB %q (Resource Group %q): %+v", name, resGroup, err)
+	}
+
+	read, err := client.Get(ctx, resGroup, serverName, name)
+	if err != nil {
+		return fmt.Errorf("Error retrieving Maria DB %q (Resource Group %q): %+v", name, resGroup, err)
+	}
+	if read.ID == nil {
+		return fmt.Errorf("Cannot read MariaDB Database %s (resource group %s) ID", name, resGroup)
+	}
+
+	d.SetId(*read.ID)
+
+	return resourceArmMariaDbDatabaseRead(d, meta)
+}
+
+func resourceArmMariaDbDatabaseRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient).mariadbDatabasesClient
+	ctx := meta.(*ArmClient).StopContext
+
+	id, err := parseAzureResourceID(d.Id())
+	if err != nil {
+		return err
+	}
+	resGroup := id.ResourceGroup
+	serverName := id.Path["servers"]
+	name := id.Path["databases"]
+
+	resp, err := client.Get(ctx, resGroup, serverName, name)
+	if err != nil {
+		if utils.ResponseWasNotFound(resp.Response) {
+			log.Printf("[WARN] MariaDB Database '%s' was not found (resource group '%s')", name, resGroup)
+			d.SetId("")
+			return nil
+		}
+
+		return fmt.Errorf("Error making Read request on Azure MariaDB Database %s: %+v", name, err)
+	}
+
+	d.Set("name", resp.Name)
+	d.Set("resource_group_name", resGroup)
+	d.Set("server_name", serverName)
+
+	if props := resp.DatabaseProperties; props != nil {
+		d.Set("charset", props.Charset)
+
+		if collation := props.Collation; collation != nil {
+			d.Set("collation", collation)
+		}
```

Comment:
I think we can remove this nil-check?

Reply:
Yep, since we already did the nil check on the properties...